src/concepts/kafka-proxies/grpc-proxy.md

```diff
@@ -1,297 +1,51 @@
+Zilla manages the synchronous request and response messages of a gRPC service on the event stream. Each message is correlated to each other with a `zilla:correlation-id` header, providing an identifier for both Zilla and Kafka workflows to act on. Correlated messages can be on the same or different Kafka topics.
 
-Here is the service we will be enhancing with Zilla and Kafka.
+## RPC Service Definitions
 
-```protobuf
-service RouteGuide {
+Zilla supports all four [gRPC service method definitions](https://grpc.io/docs/what-is-grpc/core-concepts/#service-definition). The request messages are routed to a Kafka topic. The return message(s) can be delivered to the same or a different topic. Zilla can also handle the stream upgrade when a client sends a single request, but the service expects a stream.
 
-  // simple RPC
-  rpc GetFeature(Point) returns (Feature) {}
+- **Simple/Unary RPC** - A single message is sent and will wait for the correlated response message and return it back to the caller.
 
-  // server-side streaming RPC
-  rpc ListFeatures(Rectangle) returns (stream Feature) {}
+  ```protobuf:no-line-numbers
+  rpc SayHello(HelloRequest) returns (HelloResponse);
+  ```
 
-  // client-side streaming RPC
-  rpc RecordRoute(stream Point) returns (RouteSummary) {}
+- **Server-side streaming RPC** - A single message is sent with a returned stream back to the caller. The correlated messages produced on the reply-to topic will be sent for the client to read until there are no more messages, and the stream will close.
 
-  // bidirectional streaming RPC
-  rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
-}
-```
+  ```protobuf:no-line-numbers
+  rpc LotsOfReplies(HelloRequest) returns (stream HelloResponse);
+  ```
 
-Here is the simplest configuration for declaring the gRPC service. There is a `server` defining the service and methods which exits to the `client` which can be routed to the running gRPC instance.
+- **Client-side streaming RPC** - The client sends a stream, producing all the messages on a topic. When the client finishes writing to the stream, it will wait for the correlated response message and return it back to the caller.
 
-```yaml
-grpc_server:
-  type: grpc
-  kind: server
-  options:
-    services:
-      - route_guide.proto
-  routes:
-    - when:
-        - method: routeguide.RouteGuide/*
-      exit: grpc_client
-grpc_client:
-  type: grpc
-  kind: client
-```
+  ```protobuf:no-line-numbers
+  rpc LotsOfGreetings(stream HelloRequest) returns (HelloResponse);
+  ```
 
-Define individual route handlers for each method or specify the `*` wildcard to handle all methods the same.
+- **Bidirectional streaming RPC** - Both the client and server use a read-write stream to produce and consume correlated messages.
```

Comment:
Suggest we mention Kafka representation of end-of-stream (null message) here as well.